### PR TITLE
Structure parser

### PR DESCRIPTION
**Pull request by randomguyeatspotato**

### CODE REVIEW

File: src/sceneParser.lua [hunk 1]
```diff
@@ -40,9 +41,9 @@ function SceneParser.loadScene(sceneLines, world, location, inputs)
 	end
 
 	local function spawnObject(key, ship)
-		local shipID, location, data, shipInfo, shipType = unpack(ship)
+		local shipID, location, data, appendex = unpack(ship)
```
Comment (xordspar0) [inline src/sceneParser.lua]: Is this supposed to be `appendix`?

Comment (xordspar0) [reply src/sceneParser.lua]: What is `appendix`? Optional data?

Comment (randomguyeatspotato) [reply src/sceneParser.lua]: Appendix just something distinctively different than data and isn't used for most objects. Structure is currently the only one that uses it. I am still trying to figure out the specifics.

File: src/gamesave.lua [hunk 2]
```diff
@@ -26,7 +26,7 @@ function Gamesave.save(saveName, world)
 
 	local fileContents = SceneParser.saveScene(world)
 
-	if not love.filesystem.isDirectory(saveDir) then
+	if not love.filesystem.getInfo(saveDir) then
```
Comment (xordspar0) [inline src/gamesave.lua]: You can verify that the save directory really is a directory in one command by passing a second argument (filtertype) to getInfo. See https://love2d.org/wiki/love.filesystem.getInfo

File: src/world/shot.lua [hunk 3]
```diff
@@ -36,7 +37,9 @@ function Shot:getSaveData(references)
 	local x, y = unpack(part.location)
 	local structure = part.fixture:getBody():getUserData()
 	structure = references[structure]
-	return {self.time, structure, x, y}
+	-- Debug
+	print(self.timer:time(), structure, x, y)
```
Comment (xordspar0) [inline src/world/shot.lua]: Do you intend to remove this?

**Pull request merged**